pkg/apis/aws/v1alpha1/defaults.go

```diff
@@ -55,3 +55,10 @@ func SetDefaults_MachineImage(obj *MachineImage) {
 		obj.Architecture = pointer.String(v1beta1constants.ArchitectureAMD64)
 	}
 }
+
+// SetDefaults_InstanceMetadata sets the default response hop limit for the IMDS.
+func SetDefaults_InstanceMetadata(obj *InstanceMetadata) {
+	if obj.EnableInstanceMetadataV2 && obj.HTTPPutResponseHopLimit == nil {
+		obj.HTTPPutResponseHopLimit = pointer.Int64(2)
+	}
```

Comment:
After the internal call, we have decided to only expose `httpTokens` and `httpPutResponseHopLimit` to user

Reply:
Keeping @unmarshall's comment in mind https://github.com/gardener/machine-controller-manager-provider-aws/pull/105#discussion_r1122564757
maybe we should enhance the `mcm-provider-aws` `InstanceMetadataOptions` struct so that we could re-use it here? your call though

Reply:
I don't think we should reuse it. For starters it is currently not possible to create a machine with IMDS completely disabled. At most you can disable it from inside containers/pods (which is handled by `httpTokens`). Therefore reusing the struct and would also expose `httpEndpoint` which would cause more headaches. 

It would also make updates to mcm-aws harder if we start coupling the structs (because it would require also updates to provider-aws) etc. Let each component define its interface even if for convenience here it's almost the same.